documentation/architecture/desktop_application/diagrams/invite.puml

```diff
@@ -0,0 +1,48 @@
+@startuml
+skinparam monochrome true
+skinparam defaultFontName Courier
+skinparam roundCorner 0
+skinparam wrapWidth 400
+
+class Invitation {
+  token: InvitationToken
+  expires_at: Timestamp
+  status: InvitationStatus
+  to: Email
```

Comment:
As I mentioned in another comment, at the data-modeling level `recipient_email` is optional but will always be populated for Invitations with a `service` scope, which are the only kinds of invitations the Desktop app will be especially interested in at this stage.

On the backend, we do also allow invitations to grant access to an entire Project or Space using the same mechanisms and workflows, while Service invitations are a specialized flavor with additional metadata needed for network-level traffic routing.

I don't know how in the weeds you want this diagram to be/become. If you like, I can drop the Elixir syntax of the CBOR representations involved as a reply on this comment thread.

That said, I am hesitant for the team to invest a lot of energy to document this sub-topic until I feel that the backend design has stabilized sufficiently. Right now I anticipate generating a bunch of churn or "well actually this is really more like..." discussions. Can we perhaps defer this diagram and related prose for a later PR?

Reply:
The purpose of this document is to give a conceptual model of how things work for the desktop app at a high-level, in order to discuss rules and features.
The exact data structures and how they are shared across other purposes (like a general mechanism for adding members to a space or a project) should not be part of this document.